net/dns/resolver/forwarder.go

```diff
@@ -206,19 +208,19 @@ func resolversWithDelays(ipps []netaddr.IPPort) []resolverAndDelay {
 	// per address family.
 	total := map[hostAndFam]int{}
 
-	rr := make([]resolverAndDelay, len(ipps))
-	for _, ipp := range ipps {
-		ip := ipp.IP()
-		if host, ok := knownDoH[ip]; ok {
+	rr := make([]resolverAndDelay, len(resolvers))
+	for _, r := range resolvers {
+		if host, ok := knownDoH[r.Addr]; ok {
```

Comment:
it's cheaper to keep knownDoH keyed as netaddr.IP as it was but instead just ParseIP here... the parsing doesn't allocate, but the ip.String() in the map lookup later does.